### ISSUE TITLE
urlencode function in notification scripts doesn't work with POSIX shell (Syntax error: Bad for loop variable)

### ISSUE THREAD

**Issue by sebastic** [Icinga/icinga2#6980]
Despite the notifications scripts using `#!/usr/bin/env bash` they are executed with the standard `/bin/sh` shell, which on Debian systems is dash.

When setting the Icinga Web 2 URL the notification scripts fail:
```
[2019-02-26 15:42:54 +0100] warning/PluginNotificationTask: Notification command for object 'localhost!example-service' (PID: 9453, arguments: '/etc/icinga2/scripts/mail-service-notification.sh' '-4' '127.0.0.1' '-6' '' '-b' '' '-c' '' '-d' '2019-02-26 15:42:54 +0100' '-e' 'example-service' '-f' 'icinga@localhost' '-i' 'http://localhost/icingaweb2' '-l' 'localhost' '-n' 'localhost' '-o' 'UNKNOWN: Test' '-r' 'root@localhost' '-s' 'UNKNOWN' '-t' 'Custom' '-u' 'Example Service' '-v' 'true') terminated with exit code 2, output: /etc/icinga2/scripts/mail-service-notification.sh: 60: /etc/icinga2/scripts/mail-service-notification.sh: Syntax error: Bad for loop variable
```

## Expected Behavior
Notification scripts don't fail.

## Current Behavior
Notification scripts fail.

## Possible Solution
Don't rely on bash, and use POSIX compliant variant for urlencode like:

 https://blogs.gnome.org/shaunm/2009/12/05/urlencode-and-urldecode-in-sh/

Or use perl URI::Escape for example:

```
perl -MURI::Escape -e "print uri_escape(\"$HOSTNAME\");"
```

## Steps to Reproduce (for bugs)
1. Configure notification commands to include Icinga Web 2 URL
2. Trigger a notification via Icinga Web 2 

## Context
No notifications were being sent due to this issue.

## Your Environment
icinga2 & icingaweb2 packages from Debian stretch-backports repository.

```
icinga2 - The Icinga 2 network monitoring daemon (version: r2.10.2-1)

Copyright (c) 2012-2018 Icinga Development Team (https://icinga.com/)
License GPLv2+: GNU GPL version 2 or later <http://gnu.org/licenses/gpl2.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

System information:
  Platform: Debian GNU/Linux
  Platform version: 9 (stretch)
  Kernel: Linux
  Kernel version: 4.9.0-8-amd64
  Architecture: x86_64

Build information:
  Compiler: GNU 6.3.0
  Build host: binet

Application information:

General paths:
  Config directory: /etc/icinga2
  Data directory: /var/lib/icinga2
  Log directory: /var/log/icinga2
  Cache directory: /var/cache/icinga2
  Spool directory: /var/spool/icinga2
  Run directory: /run/icinga2

Old paths (deprecated):
  Installation root: /usr
  Sysconf directory: /etc
  Run directory (base): /run
  Local state directory: /var

Internal paths:
  Package data directory: /usr/share/icinga2
  State path: /var/lib/icinga2/icinga2.state
  Modified attributes path: /var/lib/icinga2/modified-attributes.conf
  Objects path: /var/cache/icinga2/icinga2.debug
  Vars path: /var/cache/icinga2/icinga2.vars
```
* Enabled features (`icinga2 feature list`):
```
Disabled features: command compatlog debuglog elasticsearch gelf graphite influxdb livestatus opentsdb perfdata statusdata syslog
Enabled features: api checker ido-pgsql mainlog notification
```
* Icinga Web 2 version and modules (System - About): 2.6.2

Loaded modules: doc 2.6.2 & monitoring 2.6.2

* Config validation (`icinga2 daemon -C`):
```
[2019-02-26 16:02:38 +0100] information/cli: Icinga application loader (version: r2.10.2-1)
[2019-02-26 16:02:38 +0100] information/cli: Loading configuration file(s).
[2019-02-26 16:02:38 +0100] information/ConfigItem: Committing config item(s).
[2019-02-26 16:02:38 +0100] information/ApiListener: My API identity: nl-asd-tc1-mon01.mngt.as60402.net
[2019-02-26 16:02:39 +0100] information/ConfigItem: Instantiated 1 NotificationComponent.
[2019-02-26 16:02:39 +0100] information/ConfigItem: Instantiated 1 IdoPgsqlConnection.
[2019-02-26 16:02:39 +0100] information/ConfigItem: Instantiated 1 CheckerComponent.
[2019-02-26 16:02:39 +0100] information/ConfigItem: Instantiated 5 UserGroups.
[2019-02-26 16:02:39 +0100] information/ConfigItem: Instantiated 4 TimePeriods.
[2019-02-26 16:02:39 +0100] information/ConfigItem: Instantiated 15 Users.
[2019-02-26 16:02:39 +0100] information/ConfigItem: Instantiated 3 Zones.
[2019-02-26 16:02:39 +0100] information/ConfigItem: Instantiated 9 ServiceGroups.
[2019-02-26 16:02:39 +0100] information/ConfigItem: Instantiated 3483 Services.
[2019-02-26 16:02:39 +0100] information/ConfigItem: Instantiated 32 HostGroups.
[2019-02-26 16:02:39 +0100] information/ConfigItem: Instantiated 320 Hosts.
[2019-02-26 16:02:39 +0100] information/ConfigItem: Instantiated 6 NotificationCommands.
[2019-02-26 16:02:39 +0100] information/ConfigItem: Instantiated 4047 Notifications.
[2019-02-26 16:02:39 +0100] information/ConfigItem: Instantiated 1 IcingaApplication.
[2019-02-26 16:02:39 +0100] information/ConfigItem: Instantiated 1 Endpoint.
[2019-02-26 16:02:39 +0100] information/ConfigItem: Instantiated 1 FileLogger.
[2019-02-26 16:02:39 +0100] information/ConfigItem: Instantiated 15 Dependencies.
[2019-02-26 16:02:39 +0100] information/ConfigItem: Instantiated 262 CheckCommands.
[2019-02-26 16:02:39 +0100] information/ConfigItem: Instantiated 29 Comments.
[2019-02-26 16:02:39 +0100] information/ConfigItem: Instantiated 2 ApiUsers.
[2019-02-26 16:02:39 +0100] information/ConfigItem: Instantiated 1 ApiListener.
[2019-02-26 16:02:39 +0100] information/ScriptGlobal: Dumping variables to file '/var/cache/icinga2/icinga2.vars'
[2019-02-26 16:02:39 +0100] information/cli: Finished validating the configuration file(s).
```

**Comment by dgoetz** [comment]
Pullrequest https://github.com/Icinga/icinga2/pull/6780 is already merged which will make the function more POSIX compliant. Can you test the command if it fits your needs? The new version will be released with 2.11, but if you find some caveats it can be adjusted before.

**Comment by sebastic** [comment]
Confirmed fixed with the changes from eee2be6e7fd3503cf5d34ee91358ae3e93421061
